### ISSUE TITLE
Get all Forge packages working

### ISSUE THREAD

**Issue by apjanke** [octave-app/octave-app#43]
Make sure that all Octave Forge packages install and pass their tests under Octave.app.

Requires [Testify](https://github.com/apjanke/octave-testify) to conveniently run the tests on installed packages.

To check one of these packages, fire up Octave 4.4.1 or 5.2, and:
* Install and load the `testify` and `doctest` packages
* Pick a package, preferably working from the top or bottom of the list, and...
* Test it:
```
pkg install -forge <package>
pkg load <package>
runtests2 -pkg <package>
```
* (If something goes wrong with `runtests2`, report it as a bug to [Testify](https://github.com/apjanke/octave-testify/issues); don't bother working around it.)
* Note the results here. Include the package version you tested.
  * If you needed to install _any_ external dependencies that were not shipped with Octave.app, note it here.
  * And if it was a build or test failure, open up a new Issue on this repo to track it. Paste in the results of running the tests.
    * And if you want to be a dear, report the failure upstream to [GNU Octave on Savannah](https://savannah.gnu.org/bugs/?group=octave), and link to the upstream report in the issue here.
* Build failures in dependencies block testing dependents. Test failures in dependencies do _not_ block testing dependents.

TODO: Write a script that will install and test all these packages together in a semi-automated fashion.

TODO: If someone wants to be really nice, turn all the package names in these list items into links to their respective Octave Forge pages.

NOTE: All the testing here that apjanke did is on his laptop or desktop, which have big Homebrews and a bunch of Octaves installed, so they're not clean testing environments. Forge packages may well have been pulling in packages from the non-Octave.app Homebrew environment, contaminating the tests with false OKs.

###  On Octave.app 4.4.1

* [ ] arduino – blocked by `instrument-control` failure
* [x] bim – 1.1.5 builds and tests OK
* [x] bsltl – 1.3.1 builds OK; package has no tests
* [x] cgi – 0.1.2 builds OK; package has no tests
* [x] communications – 1.2.1 build failure: #147
* [x] control – 3.2.0 builds and tests OK
* [x] data-smoothing – 1.3.0 builds OK; package has no tests
* [ ] database
* [x] dataframe – 1.2.0 builds OK; package has no tests
* [ ] dicom – requires libgdcm not shipped with Octave.app; 0.2.2 build failure: #148
* [x] divand – 1.1.2 builds and tests OK (though its tests are trivial)
* [x] doctest – 0.7.0 builds OK; package has no tests
* [x] econometrics – 1.1.2 builds OK; package has no tests
* [ ] fem-fenics – blocked by Fenics installation; see #45
* [ ] financial – 0.5.3 builds OK; test failure: #149
* [ ] fits – 1.0.7 build fails: #150
* [X] fpl – 1.3.5 builds OK; package has no tests
* [ ] fuzzy-logic-toolkit – 0.4.5 build fails: #151
* [x] ga – 0.10.0 builds and tests OK
* [x] general – 2.1.0 builds and tests OK
* [x] generate_html – 0.3.1 builds OK; package has no tests
* [ ] geometry – 3.0.0 builds OK; test failure: #153
* [x] gsl – 2.1.1 build fails: #154
* [ ] image – 2.10.0 builds OK; test failure: #155
* [ ] image-acquisition – 0.2.2 build failure: #156
* [ ] instrument-control – 0.4.0 builds OK; fails tests:  #44
* [ ] interval – 3.2.0 requires mpfr  >=3.1.0 library (I think): #157
* [x] io – 2.4.13 builds and tests OK
* [ ] level-set – 0.3.0 build fails: #158
* [x] linear-algebra – 2.2.2 build fails: #159
* [x] lssa – 0.1.3 builds and tests OK
* [x] ltfat – 2.3.1 builds and tests OK
* [x] mapping – 1.2.1 builds and tests OK
* [ ] miscellaneous – 1.2.1 builds OK; fails tests: #160
* [ ] mpi – 1.2.0 build fails: #161
* [ ] msh – 1.0.10 builds OK; fails tests: #145
* [x] mvn – 1.1.0 builds and tests OK, but tests are trivial
* [x] nan – 3.1.4 builds w many warnings; fails tests: #144
* [x] ncarray – 1.0.4 builds OK; package has no tests
* [x] netcdf – 1.0.12 builds and tests OK
* [x] nurbs – 1.3.13 builds and tests OK
* [ ] ocs – 0.1.5 build fails: #162
* [x] octclip – 1.0.8 builds and tests OK
* [ ] octproj – 1.1.5 build fails: #163
* [x] optics – 0.1.4 builds and tests OK
* [x] optim – 1.6.0 builds and tests OK
* [x] optiminterp – 0.3.5 builds and tests OK
* [x] parallel – 3.1.3 builds and tests OK
* [ ] quaternion – crash when running tests! #164
* [ ] queueing – 1.2.6 - builds OK; test fails: #165
* [x] secs1d – 0.0.9 builds OK; package has no tests
* [ ] secs2d – 0.0.8 build fails: #166
* [x] secs3d – 0.0.1 builds OK; package has no tests
* [ ] signal – 1.4.1 fails tests: #146
* [x] sockets – 1.2.0 builds OK; package has no tests
* [ ] sparsersb – 1.0.6 build fails: #167
* [x] splines – 1.3.2 builds and tests OK
* [x] statistics – 1.4.1 builds and tests OK
* [x] stk – 2.6.1 builds and tests OK
* [x] strings – 1.2.0 builds and tests OK
* [x] struct – 1.0.16 builds and tests OK
* [ ] symbolic – 2.8.0 test failure: #168
* [ ] tisean – 0.2.3 build failure: #169
* [x] tsa – 4.4.5 builds and tests OK
* [x] vibes – 0.2.0 builds and tests OK
* [ ] video – 1.2.4 build failure: #170
* [ ] vrml – blocked by linear-algebra failure
* [ ] zeromq – 1.5.0 build failure: #171

##  Housekeeping

Blocks https://github.com/octave-app/octave-app-bundler/issues/60.

**Comment by schoeps** [comment]
To make this a more doable project, shall we start with the most popular packages? (where "popular" needs to be defined)

**Comment by apjanke** [comment]
Yeah, if we could prioritize them by popularity, that would be great. Do you know where to find Octave Forge download statistics? That would be a good first measurement.

**Comment by apjanke** [comment]
Oh! As part of this, we should plot out a big dependency graph of which packages depend on which, and also start with the most-depended-upon packages. I'll see if I can add that capability to Packajoozle, using GraphViz.

**Comment by apjanke** [comment]
Okay, I got through testing all the Forge packages under 4.4.1-u1 beta2. About half of them build and pass tests; half of them don't. That's... not good.

**Comment by apjanke** [comment]
Yeah, you know what, nevermind about this. I've come to view Octave Forge packages as somewhat low quality and brittle, and also mostly targeted at Linux and few people use or develop them on macOS. So, I'm no longer going to consider a Forge package build or test failure as a problem in itself. I'm only going to worry about it if I have specific user complaints, or if my tests indicate there's a failure under Octave.app that does not occur under a vanilla Homebrewed Octave.

Closing this as WontFix/Upstream.
